diff --git a/seat.c b/seat.c
index 8008dd9..7b1be09 100644
--- a/seat.c
+++ b/seat.c
@@ -126,12 +126,7 @@ update_capabilities(struct cg_seat *seat)
 	}
 	wlr_seat_set_capabilities(seat->seat, caps);

-	/* Hide cursor if the seat doesn't have pointer capability. */
-	if ((caps & WL_SEAT_CAPABILITY_POINTER) == 0) {
-		wlr_cursor_unset_image(seat->cursor);
-	} else {
-		wlr_cursor_set_xcursor(seat->cursor, seat->xcursor_manager, DEFAULT_XCURSOR);
-	}
+	wlr_cursor_unset_image(seat->cursor);
 }

 static void
@@ -483,7 +478,6 @@ handle_request_set_cursor(struct wl_listener *listener, void *data)
 	/* This can be sent by any client, so we check to make sure
 	 * this one actually has pointer focus first. */
 	if (focused_client == event->seat_client->client) {
-		wlr_cursor_set_surface(seat->cursor, event->surface, event->hotspot_x, event->hotspot_y);
 	}
 }
